### ISSUE TITLE
Low resolution object segmentation in high resolution images?

### ISSUE THREAD

**Issue by AugustLundquist** [MouseLand/cellpose#317]
Hi,

When segmenting large objects in a high resolution image, in our case approximately 300 pixel diameter objects in 2000x2000 image, the object segmentation seem to be at a much lower resolution than what would be possible (see example attached). Is this intended behaviour or could there be some scaling issue?

I run Cellpose through this notebook: https://colab.research.google.com/drive/1958UQIH-XAYogKvbxnaUHALYvR73KLj2?usp=sharing

![image1](https://user-images.githubusercontent.com/75359950/130105103-363b8b6f-eb11-4f51-b154-4c85253474b5.png)

**Comment by kevinjohncutler** [comment]
This is due to the downscaling that Cellpose does for large cells (in this case, 30/300 = 1/10, so your masks are 1/10th the effective pixel density of your image). I have an idea to solve this: run Cellpose on the downsampled image as usual, but run the dynamics on upscaled, interpolated network outputs. Currently the flow is already interpolated by default, so the novel thing here would be that the cell probability (or distance in Omnipose) is able to seed much higher resolution masks. This will make dynamics slower (maybe by 100x in your case, with each dimension is bigger by a factor of 10 when upscaled) but could be worth it for capturing detail (although you are still limited by the amount of detail that the network gives you at 1/10th the resolution). I originally thought of this approach in the context of making 'sub-pixel' masks, _i.e._ going _higher_ resolution than the input image, which could be handy for protein localization in bacteria.

**Comment by carsen-stringer** [comment]
This is a neat approach! I think upsampling to the original size of the image might be enough. The parameter for upsampling before running the dynamics is `resample=True` which you can add to your `model.eval` call, e.g. `model.eval(img, ... , resample=True)`, or on the command line you can add the flag `--resample`
